File: codes/prime.py
import math
import time

TARGET = 310

def is_prime(n):
    if n < 2:
        return False
    if n % 2 == 0:
        return n == 2

    limit = math.isqrt(n)
    d = 3

    while d <= limit:
        if n % d == 0:
            return False
        d += 2

    return True

start = time.perf_counter()
n = 2
count = 0
last_prime = 0

while time.perf_counter() - start < TARGET:
    if is_prime(n):
        count += 1
        last_prime = n
    n += 1

elapsed = time.perf_counter() - start

print("workload=prime")
print(f"tested_until={n}")
print(f"primes={count}")
print(f"last_prime={last_prime}")
print(f"elapsed={elapsed:.3f}s")

File: codes/Prime.java
public class Prime {
    static boolean isPrime(long n) {
        if (n < 2) return false;
        if (n % 2 == 0) return n == 2;

        for (long d = 3; d <= n / d; d += 2)
            if (n % d == 0)
                return false;

        return true;
    }

    public static void main(String[] args) {
        double target = 310.0;

        long start = System.nanoTime();
        long n = 2;
        long count = 0;
        long lastPrime = 0;

        while ((System.nanoTime() - start) / 1e9 < target) {
            if (isPrime(n)) {
                count++;
                lastPrime = n;
            }
            n++;
        }

        double elapsed = (System.nanoTime() - start) / 1e9;

        System.out.println("workload=prime");
        System.out.println("tested_until=" + n);
        System.out.println("primes=" + count);
        System.out.println("last_prime=" + lastPrime);
        System.out.println("elapsed=" + elapsed + "s");
    }
}